OrrRock/OrrRock/RouteFinding/RouteFindingMainViewController.swift

```diff
@@ -0,0 +1,156 @@
+                animated: true,
+                completion: nil
+            )
+        }
+    }
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        setUpLayout()
+        setNavigationBar()
+        setSegment()
+        
+    }
+    
+    override func viewWillDisappear(_ animated: Bool) {
+        self.navigationController?.isNavigationBarHidden = false
+    }
+    
+    override func viewWillAppear(_ animated: Bool) {
+        self.navigationController?.isNavigationBarHidden = true
+    }
+    
+    func setSegment(){
+        self.segmentedControl.selectedSegmentIndex = 0
+        if let firstVC = dataViewControllers.first {
+            pageViewController.setViewControllers([firstVC], direction: .forward, animated: true, completion: nil)
+        }
+        self.segmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.orrGray400,.font: UIFont.systemFont(ofSize: 17, weight: .bold)], for: .normal)
+        self.segmentedControl.setTitleTextAttributes(
+            [
+                NSAttributedString.Key.foregroundColor: UIColor.orrBlack,
+                .font: UIFont.systemFont(ofSize: 17, weight: .bold)
+            ],
+            for: .selected
+        )
+        self.segmentedControl.addTarget(self, action: #selector(changeValue(control:)), for: .valueChanged)
+        self.segmentedControl.selectedSegmentIndex = 0
+        self.changeValue(control: self.segmentedControl)
+    }
+    
+    @objc private func changeValue(control: UISegmentedControl) {
+        self.currentPage = control.selectedSegmentIndex
+    }
+    
+    func setUpLayout(){
+        view.backgroundColor = .orrGray100
+        view.addSubview(topView)
+        topView.snp.makeConstraints {
+            $0.top.equalTo(view.forLastBaselineLayout.snp_topMargin).offset(OrrPd.pd8.rawValue)
+            $0.width.equalToSuperview()
+            $0.height.equalTo(26)
+        }
+        topView.addSubview(titleLabel)
+        titleLabel.snp.makeConstraints {
+            $0.leading.equalToSuperview().offset(OrrPd.pd16.rawValue)
+        }
+        topView.addSubview(plusButton)
+        plusButton.snp.makeConstraints{
+            $0.trailing.equalToSuperview().offset(-OrrPd.pd16.rawValue)
+        }
+        
+        view.addSubview(segmentedControl)
+        segmentedControl.snp.makeConstraints {
+            $0.width.equalToSuperview()
+            $0.top.equalTo(topView.snp.bottom).offset(OrrPd.pd16.rawValue)
+            $0.height.equalTo(30)
+        }
+        addChild(pageViewController)
+        view.addSubview(pageViewController.view)
+        pageViewController.view.snp.makeConstraints {
+            $0.top.equalTo(segmentedControl.snp.bottom)
+            $0.leading.trailing.bottom.equalToSuperview()
+        }
+        pageViewController.didMove(toParent: self)
+        
+    }
+    
+    func setNavigationBar(){
+        self.navigationController?.isNavigationBarHidden = true
+    }
```

Comment:
```suggestion
    func setNavigationBarHidden(_ Boolean: Bool) {
        if Boolean = true {
            self.navigationController?.isNavigationBarHidden = true
        }
        else {
            self.navigationController?.isNavigationBarHidden = false
        }
    }
```

이렇게 모듈화를 해서 91번째줄과 95번째 줄에서 둘다 활용해 보는 것은 어떨까요?

Reply:
setNavigationBar라는 함수는 setUpLayout처럼 뷰가 처음 viewdidload를 호출할시에 뷰와 관련된 설정들을 초기화해주는 함수로 사용하고 있었고, 이 함수는 네비게이션 바와 관련된 설정들을 담당하고 있었습니다. 따라서 해당 함수는 한번만 호출이 되어야 하고 ,해당 함수에 파라미터를 받아 파라미터의 값에 따른 분기처리는 필요하지 않을 것 같습니다. 
다만 이번의 setNavigationBar 라는 네이밍은 네비게이션 바 설정 관련 함수가 한가지 뿐이라 보는 사람으로 하여금 혼란이 생길것 같네요! 수정하도록 하겠습니다.